src/Pim/Bundle/CatalogBundle/ElasticSearch/Filter/StringFilter.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\ElasticSearch\Filter;
+
+use Akeneo\Component\StorageUtils\Exception\InvalidPropertyTypeException;
+use Pim\Component\Catalog\Model\AttributeInterface;
+use Pim\Component\Catalog\Query\Filter\AttributeFilterInterface;
+use Pim\Component\Catalog\Query\Filter\Operators;
+use Pim\Component\Catalog\Validator\AttributeValidatorHelper;
+
+/**
+ * String Filter for an ElasticSearch query
+ *
+ * @author    Samir Boulil <samir.boulil@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class StringFilter extends AbstractFilter implements AttributeFilterInterface
+{
+    /**
+     * @param AttributeValidatorHelper $attrValidatorHelper
+     * @param array                    $supportedAttributeTypes
+     * @param array                    $supportedOperators
+     */
+    public function __construct(
+        AttributeValidatorHelper $attrValidatorHelper,
+        array $supportedAttributeTypes = [],
+        array $supportedOperators = []
+    ) {
+        $this->attrValidatorHelper = $attrValidatorHelper;
+        $this->supportedAttributeTypes = $supportedAttributeTypes;
+        $this->supportedOperators = $supportedOperators;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addAttributeFilter(
+        AttributeInterface $attribute,
+        $operator,
+        $value,
+        $locale = null,
```

Comment:
I respected the signature declared by the interface: 
https://github.com/jjanvier/pim-community-dev/blob/017349e799d2d460f11b9e57f090ba77dd4682e8/src/Pim/Component/Catalog/Query/Filter/AttributeFilterInterface.php#L32-L32

So my guess is, we keep it.